### PR TITLE
show project download info at UI

### LINKED ISSUE #3508: There should be terminal output when downloading a project using --starter
/kind feature

## Which functionality do you think we should add?

When downloading a project using `--starter` we should output in the terminal / logging that the project has actually been downloaded.

For example:
```
▶ odo create nodejs --starter
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [153046ns]
 ✓  Checking devfile compatibility [223245ns]
 ✓  Creating a devfile component from registry: DefaultDevfileRegistry [274307ns]
 ✓  Validating devfile component [718797ns]

Please use `odo push` command to create the component with source deployed
```

Actually has no output that it's been downloaded. Ideally, it should look something like this:

```sh
▶ odo create nodejs --starter
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [153046ns]
 ✓  Checking devfile compatibility [223245ns]
 ✓  Creating a devfile component from registry: DefaultDevfileRegistry [274307ns]
 ✓  Validating devfile component [718797ns]

Project
 ✓  Downloaded project from github.com/odo-devfiles/nodejs [3s]

Please use `odo push` command to create the component with source deployed
```

## Why is this needed?

We should output what we have actually done / completed.

### PR DESCRIPTION
**Pull request by jichenjc**
**What type of PR is this?**
show project download info at UI

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3508 

**How to test changes / Special notes to the reviewer**:
`odo create  --starter nodejs`

### CODE REVIEW

File: pkg/odo/cli/component/create.go [hunk 1]
```diff
@@ -910,12 +910,16 @@ func (co *CreateOptions) downloadProject(projectPassed string) error {
 		return errors.Errorf("Project type not supported")
 	}
 
+	log.Info("Project")
+	downloadSpinner := log.Spinnerf("Downloaded project from %s", url)
```
Comment (mik-dass) [inline pkg/odo/cli/component/create.go:914]: `Downloading`

Comment (jichenjc) [reply pkg/odo/cli/component/create.go]: ok, thanks for the comments, updated

File: pkg/odo/cli/component/create.go [hunk 2]
```diff
@@ -910,12 +910,16 @@ func (co *CreateOptions) downloadProject(projectPassed string) error {
 		return errors.Errorf("Project type not supported")
 	}
 
+	log.Info("Project")
```
Comment (cdrage) [inline pkg/odo/cli/component/create.go:913]: You need to add `\n` within this. See output, as there is no spacing between Validation and Project

```sh
~/openshift/f3                                                                                                                                                                                                                                                                   
▶ odo create --starter nodejs
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [155676ns]
 ✓  Checking devfile compatibility [452063ns]
 ✓  Creating a devfile component from registry: DefaultDevfileRegistry [323241ns]
 ✓  Validating devfile component [357302ns]
Project
 ✓  Downloading project from https://codeload.github.com/odo-devfiles/nodejs-ex/legacy.zip/master [294ms]

Please use `odo push` command to create the component with source deployed
```

**Comment by cdrage** [comment]
Also, the output seems incorrect? The URL is straight from GitHub's CDN rather than the actual repo which would be: `https://github.com/odo-devfiles/nodejs-ex`

```sh
~/openshift/f3                                                                                                                                                                                                                                                                   
▶ odo create --starter nodejs
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [155676ns]
 ✓  Checking devfile compatibility [452063ns]
 ✓  Creating a devfile component from registry: DefaultDevfileRegistry [323241ns]
 ✓  Validating devfile component [357302ns]
Project
 ✓  Downloading project from https://codeload.github.com/odo-devfiles/nodejs-ex/legacy.zip/master [294ms]

Please use `odo push` command to create the component with source deployed
```

File: pkg/odo/cli/component/create.go (hunk 2 above)
Comment (jichenjc) [reply pkg/odo/cli/component/create.go]: updated

**Comment by jichenjc** [comment]
> 
> 
> Also, the output seems incorrect? The URL is straight from GitHub's CDN rather than the actual repo which would be: `https://github.com/odo-devfiles/nodejs-ex`
> 
> ```shell
> ~/openshift/f3                                                                                                                                                                                                                                                                   
> ▶ odo create --starter nodejs
> Experimental mode is enabled, use at your own risk
> 
> Validation
>  ✓  Checking devfile existence [155676ns]
>  ✓  Checking devfile compatibility [452063ns]
>  ✓  Creating a devfile component from registry: DefaultDevfileRegistry [323241ns]
>  ✓  Validating devfile component [357302ns]
> Project
>  ✓  Downloading project from https://codeload.github.com/odo-devfiles/nodejs-ex/legacy.zip/master [294ms]
> 
> Please use `odo push` command to create the component with source deployed
> ```

will use a follow up to check why this is using legacy code ..

**Comment by jichenjc** [comment]
ping @cdrage can you help to check above comment? THanks

**Comment by cdrage** [comment]
Hey @jichenjc the preference would be the GitHub repo URL / name. Having the CDN / zip file location I think would be a bad idea.

**Comment by jichenjc** [comment]
> Hey @jichenjc the preference would be the GitHub repo URL / name. Having the CDN / zip file location I think would be a bad idea.

works for me :)

**Comment by jichenjc** [comment]
```
]$ odo create --starter nodejs
Experimental mode is enabled, use at your own risk

Validation
 ✓  Creating a devfile component from devfile path: /home/jichen/t/devfile.yaml [88426ns]
 ✓  Validating devfile component [112528ns]

Project
 ✓  Downloading project from https://github.com/odo-devfiles/nodejs-ex.git [497ms]

Please use `odo push` command to create the component with source deployed
```

@cdrage updated to this , thanks for your comments

**Comment by jichenjc** [comment]
ping @cdrage ,can you help another review? Thanks

**Pull request merged**